app/searchers/responses_searcher.rb

```diff
@@ -232,12 +240,10 @@ def filter_by_questions(qualifier_text, token_values)
     question_code = qualifier_text[1..-2]
     matched_question = Question.where("LOWER(code) = ?", question_code.downcase).first
     return false if matched_question.blank?
-    matched_qings = Questioning.where(question: matched_question)
-    return false if matched_qings.blank?
-    # Get the deterministic Questioning ID from the Question
-    matched_qing_id = matched_qings.filter_unique.first.id
+    matched_qing_ids = Questioning.where(question: matched_question).pluck(:id)
```

Comment:
Why pluck the IDs here and then redo this query moments later? Could you just store the relation in the `qings` array instead?

Reply:
Also a big fat comment seems worthwhile here explaining that this is an intermediate result and will be further refined in a second parsing pass once the form filter has been parsed or wahtever.